(treeData): add sort to Tree Data

ghiscoding opened the pull request.

Alex-ley commented:
I’ll pull this tomorrow and see what’s going on :)

Excited to see how filtering and aggregators might work!

ghiscoding commented:
Hey @Alex-ley 

I added the string/numeric sort functions and I see that it's not fully working and I understand why but I think it's better to wait for the Aggregators to be finished before re-looking at the sorting again. So basically if you look at the image below, you can see that the `size` is sorted correctly in each sub-array (children) but anything outside of it isn't considered in the sort. The reason is because of the fact that we use recursion and also because when we are in a parent, then the size is not in the `size` item property but instead is in the `__agg__size` and so because of that it kinda sees that there's nothing to sort and moves to the next row, which explains why every sub-array (children) are sorted (because they have the `size` property filled)... so again I think it's better to get the Aggregators working first, then rethink about the sorting later.

With that said, I'll merge as is and let you start looking at the Aggregators (just make sure to pull my code)...

oh and you might be interested to know about this [change](https://github.com/ghiscoding/SlickGrid/pull/7/files#diff-9b094a485b187eb9ad072033ac4bd63eR141) as well, I added a `type` in each column definition (string, number, date) to know which sort method to use when sorting. I'm not sure if that could be useful for your code, but I use such `type` in my other libs (for Sorting, Formatters and some other reasons that I forgot) so it's nothing really new in my lib but good to know about.

![image](https://user-images.githubusercontent.com/643976/86838553-526c8280-c06e-11ea-89f2-4177f063bde2.png)
